dms-infrastructure/src/main/kotlin/team/aliens/dms/thirdparty/parser/ExcelAdapter.kt

```diff
@@ -0,0 +1,78 @@
+package team.aliens.dms.thirdparty.parser
+
+import com.fasterxml.uuid.Generators
+import java.io.File
+import org.apache.poi.hssf.usermodel.HSSFWorkbook
+import org.apache.poi.ss.usermodel.Row.CREATE_NULL_AS_BLANK
+import org.apache.poi.ss.usermodel.Workbook
+import org.apache.poi.xssf.usermodel.XSSFWorkbook
+import org.springframework.stereotype.Component
+import team.aliens.dms.domain.file.FileExtension.XLS
+import team.aliens.dms.domain.file.FileExtension.XLSX
+import team.aliens.dms.domain.student.model.Sex
+import team.aliens.dms.domain.student.model.VerifiedStudent
+import team.aliens.dms.domain.file.spi.ParseFilePort
+import team.aliens.dms.thirdparty.parser.exception.ExcelExtensionMismatchException
+import team.aliens.dms.thirdparty.parser.exception.ExcelInvalidFileException
+import team.aliens.dms.thirdparty.parser.exception.ExcelSexMismatchException
+
+@Component
+class ExcelAdapter(
+
+) : ParseFilePort {
+
+    override fun transferToVerifiedStudent(file: File): List<VerifiedStudent> {
+        val workbook = transferToExcel(file)
+
+        val verifiedStudents = mutableListOf<VerifiedStudent>()
+
+        try {
+            val worksheet = workbook.getSheetAt(0)
+
+            for (i in 1..worksheet.lastRowNum) {
+                val excelData = worksheet.getRow(i).run {
+                    VerifiedStudent(
+                        id = Generators.timeBasedGenerator().generate(),
+                        schoolName = getCell(0, CREATE_NULL_AS_BLANK).stringCellValue,
+                        name = getCell(1, CREATE_NULL_AS_BLANK).stringCellValue,
+                        gcn = getCell(2, CREATE_NULL_AS_BLANK).numericCellValue.toInt().toString(),
+                        roomNumber = getCell(3, CREATE_NULL_AS_BLANK).numericCellValue.toInt(),
+                        sex = transferToSex(
+                            getCell(4, CREATE_NULL_AS_BLANK).stringCellValue
+                        )
+                    )
+                }
+
+                verifiedStudents.add(excelData)
+            }
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
+
+
+        return verifiedStudents
+    }
+
+    private fun transferToExcel(file: File): Workbook {
+        val inputStream = file.inputStream()
+
+        return runCatching {
+            when (file.extension.lowercase()) {
+                XLS -> HSSFWorkbook(inputStream)
+                XLSX -> XSSFWorkbook(inputStream)
+                else -> throw ExcelExtensionMismatchException
+            }
+        }.also {
```

Comment:
p3; use 메서드를 사용하는 건 어떤가요?

Reply:
이유가 궁금합니다

Reply:
use 메서드가 Closable을 상속받아 자동으로 inputStream을 close하는 걸 알고 계시나요?
굳이 inputStream.close할 필요 없이 사용할 수 있어 가독성 측면에서도 좋을 것 같습니다

Reply:
inputstream 하나를 close 하는데  use를 사용할만한 이유를 잘 모르겠습니다. 오히려 불필요한 it을 만들어서 성능이 아주 조금 안좋아 지지 않을까요?

Reply:
음.. it 키워드로 인해 성능이 저하 되나요?

Reply:
그건 키워드를 사용하지 않아서 그렇지 않을거 같아요😅
inputstream을 close하기 위해서 블록을 사용하는게 가독성이 좋은지가 의문입니다 블록안에 블록이 여러게 있으면 오히려 가독성이 안좋아 보이는데... 판단해주시면 그렇게 하겠습니다.

Reply:
블록 안에 블록 만드는게 아닙니다..
아래와 같이 바꾸면 되는 거에요..
```suggestion
        }.use {
```

Reply:
저건 불가능합니다

Reply:
? 됩니다

Reply:
runblocking 블록에서는 use가 사용 불가능합니다 inputstream 을 블록으로 만들어야 가능하죠

Reply:
아 제가 코드를 잘못 수정했네요

`file.inputStream().use { }`

이런 구문을 생각했어요

Reply:
이제 더이상 수정 없습니다 지금 이대로 가시죠